schema-registry/rest-service/src/main/java/com/hortonworks/registries/schemaregistry/webservice/SchemaRegistryResource.java

```diff
@@ -606,6 +608,179 @@ public Response getSchemaVersionById(@ApiParam(value = "version identifier of th
         return response;
     }
 
+    @GET
+    @Path("/schemas/versionsById/states")
+    @ApiOperation(value = "Get schema version life cycle states",
+            response = SchemaVersionInfo.class, tags = OPERATION_GROUP_SCHEMA)
+    @Timed
+    public Response getSchemaVersionLifeCycleStates() {
+        Response response;
+        try {
+            List<SchemaVersionLifecycleState> states = schemaRegistry.getSchemaVersionLifecycleStates();
+            response = WSUtils.respondEntity(states, Response.Status.OK);
+        } catch (Exception ex) {
+            LOG.error("Encountered error while getting schema version lifecycle states", ex);
+            response = WSUtils.respond(Response.Status.INTERNAL_SERVER_ERROR, CatalogResponse.ResponseMessage.EXCEPTION, ex.getMessage());
+        }
+
+        return response;
+    }
+
+    @POST
+    @Path("/schemas/versionsById/{id}/states/enable")
```

Comment:
why is states pluralised, surely a schema version can have only a singular state.

e.g. should this not be:
/schemas/versionsById/{id}/state/enable